app/src/main/java/com/example/splendormobilegame/activities/GameActivity/DeckReservingController.java

```diff
@@ -44,11 +63,25 @@ public class ReservationFromDeckMessageHandler extends UserReaction {
         public UserMessage react(ServerMessage serverMessage) {
             Log.i("UserReaction", "Entered ReservationFromDeckMessageHandler react method");
 
-            // TODO Update the model
+            
+            MakeReservationFromDeck.ResponseData responseData = (MakeReservationFromDeck.ResponseData) ReactionUtils.getResponseData(serverMessage, MakeReservationFromDeck.ResponseData.class);
+
+
+            User user=Model.getRoom().getUserByUuid(Model.getUserUuid());
+            Card card=Model.getRoom().getGame().getCardByUuid(responseData.card.uuid);
+            ReservedCard reservedCard= new ReservedCard(card,user,true );
```

Comment:
By the rules of game it should not be visible for other players. Set this visibility parameter to False.